pkg/cloudprovider/providers/azure/azure_cache.go

```diff
@@ -19,63 +19,138 @@ package azure
 import (
 	"sync"
 	"time"
-
-	"k8s.io/client-go/tools/cache"
 )
 
-type timedcacheEntry struct {
-	key  string
-	data interface{}
-}
+type getFunc func(key string) (interface{}, error)
+type listFunc func() (map[string]interface{}, error)
+
+// timedCache is a cache with TTL.
+type timedCache struct {
+	store       map[string]interface{}
+	lock        sync.Mutex
+	ttl         time.Duration
+	lastRefresh time.Time
 
-type timedcache struct {
-	store cache.Store
-	lock  sync.Mutex
+	getter getFunc
+	lister listFunc
 }
 
-// ttl time.Duration
-func newTimedcache(ttl time.Duration) timedcache {
-	return timedcache{
-		store: cache.NewTTLStore(cacheKeyFunc, ttl),
+// newTimedcache creates a new timedCache with TTL.
+func newTimedcache(ttl time.Duration, getFunc getFunc, listFunc listFunc) *timedCache {
```

Comment:
can we check nil against getFunc, listFunc return error if any of them is nil? We shouldn't wait to Get/List to check against nil

Reply:
ACK, good catch